### PR TITLE
Support creating vpa resources

### PR DESCRIPTION
**Pull request by DebakelOrakel**
## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/main.jsonnet [hunk 1]
```diff
@@ -18,6 +18,36 @@ local namespace = kube.Namespace(params.namespace) {
...
+      targetRef: {
+        apiVersion: 'apps/v1',
+        kind: std.get(vpa, 'kind', 'Deployment'),
+        name: namespacedName(name, '').name,
+      },
+      updatePolicy: {
+        updateMode: std.get(vpa, 'mode', 'Off'),
+      },
+    },
+  }
```
Comment (glrf) [inline component/main.jsonnet:47]: For something like this I would recommend that you expose the complete spec in the config hierarchy.  Something like:

```
autoscaler:
  foo/bar:
    kind: StatefulSet
    mode: Auto
    spec: # can contain arbitrary autoscaler spec
      resourcePolicy:
        containerPolicies:
          - containerName: '*'
            minAllowed:
              cpu: 100m
              memory: 50Mi
            maxAllowed:
              cpu: 1
              memory: 500Mi
            controlledResources: ["cpu", "memory"]     
```

That way you have a simple config for most cases, but for future special cases you won't need to update the component all the time.

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 2]
```diff
@@ -10,11 +10,36 @@ default:: `syn-vertical-pod-autoscaler`
...
 A dictionary of images used by this component.
+
+== `autoscaler`
+
+[horizontal]
+type:: string
+default:: {}
+
+Create autoscaler resources.
+
```
Comment (glrf) [inline docs/modules/ROOT/pages/references/parameters.adoc:28]: Please add a quick explanation how the key is used and what fields are available.

**Review by glrf — COMMENTED** [review]
Looks pretty good overall.

However please at least add some example autoscaler to the `tests/default.yml` (or maybe add a `tests/autoscaler.yml`) so changes to this show up in the golden tests.

File: docs/modules/ROOT/pages/references/parameters.adoc (hunk 2 above)
Comment (DebakelOrakel) [reply docs/modules/ROOT/pages/references/parameters.adoc:28]: I added an example under that topic.

File: component/main.jsonnet (hunk 1 above)
Comment (DebakelOrakel) [reply component/main.jsonnet]: done

**Review by glrf — APPROVED** [review]

**Pull request merged**